prealloc.go

```diff
@@ -215,13 +226,32 @@ func (v *returnsVisitor) Visit(node ast.Node) ast.Visitor {
 					if !ok {
 						continue
 					}
-					if genD.Tok == token.VAR {
+					if genD.Tok == token.TYPE {
+						for _, spec := range genD.Specs {
+							tSpec, ok := spec.(*ast.TypeSpec)
+							if !ok {
+								continue
+							}
+
+							if _, ok := tSpec.Type.(*ast.ArrayType); ok {
+								arrayTypes = append(arrayTypes, tSpec.Name.Name)
+							}
+						}
+					} else if genD.Tok == token.VAR {
 						for _, spec := range genD.Specs {
 							vSpec, ok := spec.(*ast.ValueSpec)
 							if !ok {
 								continue
 							}
-							if /*arrayType*/ _, ok := vSpec.Type.(*ast.ArrayType); ok {
+							var isArrType bool
+							switch vSpec.Type.(type) {
```

Comment:
Can we assign a value on the switch as well? Then we won't need the type assert on 251 :+1: 

```Go
switch val := vSpec.Type.(type) {
case *ast.ArrayType:
	isArrType = true
case *ast.Ident:
	isArrType = contains(arrayTypes, val.Name)
}


```

Reply:
done. 👍